File: src/app/core/types/order.ts
export class Order {
    id: number;
    items: any[];
    note: string;
    status: string;
    buyerId: number;
    date: Date;
    estimatedDate: Date;
    rejecetedNote: string;
    createdDate: Date;

    constructor(items: any[], note: string, buyerId: number, date: Date) {
        this.items = items;
        this.note = note;
        this.buyerId = buyerId;
        this.date = date;
    }
}

File: backend/Marbomil.Infrastructure/Entities/Database/Order.cs
using Marbomil.Infrastructure.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Marbomil.Infrastructure.Entities.Database
{
    public class Order : IDbEntity
    {
        public int Id { get; set; }
        
        public int? BuyerId { get; set; }
        [ForeignKey("BuyerId")]
        public virtual User Buyer { get; set; }

        public DateTime Date { get; set; }
        public string Note { get; set; }
        public OrderStatus Status { get; set; }

        public virtual ICollection<OrderItem> Items { get; set; }
    }
}
